Sources/MapboxMapsSwiftUI/Map.swift

```diff
@@ -0,0 +1,152 @@
+@_exported import MapboxMaps
+import SwiftUI
+
+@_spi(Experimental)
+@available(iOS 13.0, *)
+public extension CameraState {
+    /// Initializes CameraState with center and zoom.
+    init(center: CLLocationCoordinate2D, zoom: CGFloat) {
+        self.init(center: center, padding: .zero, zoom: zoom, bearing: 0, pitch: 0)
+    }
+}
+
+/// The action that is called when the map is loaded.
+@_spi(Experimental)
+@available(iOS 13.0, *)
+public typealias MapLoadedAction = (MapboxMap) -> Void
+
+@_spi(Experimental)
+@available(iOS 13.0, *)
+public typealias Map = InternalMap
+
+/// A view displaying Mapbox Map in SwiftUI.
+@_spi(Experimental)
+@available(iOS 13.0, *)
+// TODO: Wrap it in Map and make internal.
+public struct InternalMap: UIViewRepresentable {
+    public typealias InitialOptionsProvider = () -> MapInitOptions
+    public typealias TapAction = (CGPoint) -> Void
+    public typealias TapQueryAction = (CGPoint, (Result<[QueriedFeature], Error>)) -> Void
+
+    typealias TapActionWithQueryPair = (options: RenderedQueryOptions?, action: TapQueryAction)
+    struct Actions {
+        var onMapLoaded: MapLoadedAction?
+        var onMapTapGesture: TapAction?
+        var tapActionsWithQuery = [TapActionWithQueryPair]()
```

Comment:
This seems to be quite tailored to a particular use case of QRF on tap, this should already be achievable for the user by hooking in tap gesture and performing QRF themselves:
```swift
.onTapGesture { location in
    mapView.mapboxMap.queryRenderedFeatures(with: location, options: options) { result in
          model.property = result
    }
```
Is there a need for a dedicated action for this?

Reply:
Capturing from the offline convo: We might need more extensive event system which would extend beyond the traditional map events, this is an instance of such event.